src/Countries/Vietnam.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+use DateTimeZone;
+use IntlDateFormatter;
+use Spatie\Holidays\Calendars\ChineseCalendar;
+
+class Vietnam extends Country
+{
+    use ChineseCalendar;
+
+    public function countryCode(): string
+    {
+        return 'vn';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            // New Year's Day
+            'Tết Dương Lịch' => '01-01',
+            // Reunification Day
+            'Ngày Giải Phóng Miền Nam, Thống Nhất Đất Nước' => '04-30',
+            // Labour Day
+            'Ngày Quốc Tế Lao Động' => '05-01',
+            //A day before the Independence Day
+            'Ngày Trước Quốc Khánh' => '09-01',
+            // Independence Day
+            'Ngày Quốc Khánh' => '09-02',
+            //A day after the Independence Day
+            'Ngày Sau Quốc Khánh' => '09-03',
```

Comment:
If you do so, the `allHolidays` function will confirm that the National Day holiday calendar includes the three days from 09/01-09/03. This seems wrong. It should only cover 2 days.

Vietnam National Day only has 2 days off (Including National Day and 1 day immediately before or after). I admit that determining this date will be complicated because it changes from year to year. But it also needs to be reprocessing.

Reply:
Just to clarify, according to [this article][1], we started to have an extra day for the Independence Festival, so we got quite a few examples to work out some logics for this extra day.

I reckon this extra day will be scheduled in a way to make the whole span a long weekend:

- If Independence Day (2/9) is Monday, the extra holiday will be Tuesday (3/9) (example in [2024][5])
- If Independence Day is Tuesday, the extra holiday will be Monday (1/9)
- If Independence Day is Wednesday, the extra holiday could be Tuesday (1/9)
- If Independence Day is Thursday, the extra holiday could be Friday (3/9) (example in [2021][2] and [2022][3])
- If Independence Day is Friday, the extra holiday could be Thursday (1/9)
- If Independence Day is Saturday (will be compensated next Monday), the extra holiday could be Friday (1/9) (example in [2023][4])
- If Independence Day is Sunday (will be compensated next Monday), the extra holiday could be Tuesday (4/9)

Before 2021, no extra day.

To me this is very subjective, this is the best we can do now.

[1]: https://ldld.quangbinh.gov.vn/3cms/Ban-in-507.htm?art=1625563722710
[2]: https://tiengiang.gov.vn/chi-tiet-tin?/thong-bao-thoi-gian-nghi-le-quoc-khanh-nuoc-cong-hoa-xa-hoi-chu-nghia-viet-nam-/32712361#:~:text=Nh%C6%B0%20v%E1%BA%ADy%2C%20d%E1%BB%8Bp%20ngh%E1%BB%89%20L%E1%BB%85,l%C3%A0%20ng%C3%A0y%20ngh%E1%BB%89%20h%C3%A0ng%20tu%E1%BA%A7n.
[3]: https://www.laocai.gov.vn/tin-thoi-su/lich-nghi-le-quoc-khanh-2-9-2022-nguoi-lao-dong-duoc-nghi-4-ngay-940271
[4]: https://thuvienphapluat.vn/phap-luat-doanh-nghiep/bai-viet/lich-nghi-quoc-khanh-2023-voi-can-bo-cong-chuc-vien-chuc-nguoi-lao-dong-3343.html
[5]: https://thuvienphapluat.vn/hoi-dap-phap-luat/839ED0F-hd-lich-nghi-le-quoc-khanh-29-nam-2024.html

Reply:
@kevinb1989 I have to admit that it is difficult to accurately calculate all holidays in Vietnam. But it's great the way it's being handled now. Thank you!

Reply:
@tanhongit thanks mate, I appreciate that 🙌